tests/Yaapii.Atoms.Tests/Map/MapNoNullsTest.cs

```diff
@@ -0,0 +1,319 @@
+// MIT License
+//
+// Copyright(c) 2017 ICARUS Consulting GmbH
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using System.Threading;
+using Xunit;
+using Yaapii.Atoms.Scalar;
+
+namespace Yaapii.Atoms.Map.Tests
+{
+    public class MapNoNullTest
+    {
+        [Fact]
+        public void ContainsWithExistingItem()
+        {
+            var map = new MapNoNulls<int, int>(
+                new Dictionary<int, int> {
+                    {0, 0}
+                }
+            );
+            Assert.True(map.Contains(new KeyValuePair<int, int>(0, 0)));
+        }
+
+        [Fact]
+        public void ContainsWithMissingItemKey()
+        {
+            var map = new MapNoNulls<int, int>(
+                new Dictionary<int, int> {
+                    {0, 0}
+                }
+            );
+            Assert.False(map.Contains(new KeyValuePair<int, int>(1, 0)));
+        }
+
+        [Fact]
+        public void ContainsWithMissingValue()
+        {
+            var map = new MapNoNulls<int, int>(
+                new Dictionary<int, int> {
+                    {0, 0}
+                }
+            );
+            Assert.False(map.Contains(new KeyValuePair<int, int>(0, 1)));
+        }
+
+        [Fact]
+        public void ContainsWithMissingItem()
+        {
+            var map = new MapNoNulls<int, int>(
+                new Dictionary<int, int> {
+                    {0, 0}
+                }
+            );
+            Assert.False(map.Contains(new KeyValuePair<int, int>(1, 1)));
+        }
+
+        [Fact]
+        public void ContainsWithNullKey()
+        {
+            var map = new MapNoNulls<object, object>(
+                new Dictionary<object, object> {
+                    {0, 0}
+                }
+            );
+            Assert.Throws(typeof(ArgumentNullException), 
```

Comment:
`Assert.Throws` has a generic version an therefore takes a generic type parameter. This makes the code mode readable 
```csharp
Assert.Throws<ArgumentNullException>(() => 
        map.Contains(new KeyValuePair<object, object>(null, 0))
);
```

This should be applied to the other tests checking for an exception as well.